app/js/views/reminders/edit-dialog.jsx

```diff
@@ -0,0 +1,144 @@
+import React from 'components/react';
+import moment from 'components/moment';
+
+export default class EditDialog extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      display: false,
+      id: null,
+      reminder: null,
+      action: null,
+      due: null,
+    };
+
+    this.server = props.server;
+    this.refreshReminders = props.refreshReminders;
+
+    this.dueDateInput = null;
+    this.dueTimeInput = null;
+
+    this.onChangeAction = this.onChangeAction.bind(this);
+    this.onChangeDue = this.onChangeDue.bind(this);
+    this.onSave = this.onSave.bind(this);
+    this.onClose = this.onClose.bind(this);
+  }
+
+  onChangeAction(evt) {
+    const action = evt.target.value;
+    this.setState({ action });
+  }
+
+  onChangeDue() {
+    const date = this.dueDateInput.value.split('-');
+    const time = this.dueTimeInput.value.split(':');
+
+    const dueMoment = moment()
+      .year(date[0]).month(date[1] - 1).date(date[2])
+      .hour(time[0]).minute(time[1]).second(0);
+    const due = Number(dueMoment.toDate());
+    this.setState({ due });
+  }
+
+  onSave() {
+    const reminder = {
```

Comment:
Just to build the reminder object in 1 line, but just realise that you have more attributes in a state object.